message_tagging_service/utils.py

```diff
@@ -29,19 +29,26 @@
 logger = logging.getLogger(__name__)
 
 
-def retrieve_modulemd_content(module_build_id):
-    """Retrieve and return modulemd.txt from MBS
+def load_module_md(module_build_id):
+    """Load corresponding module metadata of a specific module build
+
+    For handling the rule match easily, this function also inject some
+    necessary module build properties into the loaded module metadata
+    mapping.
 
     :param int module_build_id: module build ID.
-    :return: modulemd content.
-    :rtype: str
+    :return: a mapping representing the module metadata.
+    :rtype: dict
     """
     api_url = conf.mbs_api_url.rstrip('/')
     resp = requests.get(f'{api_url}/module-builds/{module_build_id}', params={
         'verbose': True
     })
     resp.raise_for_status()
-    return resp.json()['modulemd']
+    module_build = resp.json()
+    modulemd = yaml.safe_load(module_build['modulemd'])
+    modulemd['data']['scratch'] = module_build['scratch']
```

Comment:
We should probably document that creating a rule with `scratch: yes` will never work since MBS does not create a build in Brew in that case.

Reply:
Good point. Your comment reminds me that what does it mean for MBS and release engineers by defining a rule including `scratch` even if it is set to `no`. Seems we should remove `scratch` from rule definition. What would be the cases of defining a rule with `scratch` included whatever the value is?

Reply:
I think at some point there were talks about creating builds in Brew for scratch builds as well. I think this was supposed to address that. We never got around to actually doing that, and it's unclear if it'll ever happen.

+1 for removing it from any rule definition, and docs.

We still need to make MTS ignore scratch builds.

Reply:
Hi @lcarva I create another PR #79. If that is accepted, we can close this one.